scripts/gen_observations.py

```diff
@@ -23,6 +27,18 @@
 out_rel_path = f'../outputs/observations/{args.obs_id}.{args.num_obs}'
 out_path = os.path.join(script_dir, out_rel_path)
 
+# Checking if observations already exist from an earlier script run
+if (os.path.isfile(os.path.join(out_path, 'drawn_nfw_profiles.npy'))):
+    # Regenerating observations (continuing script)
+    if args.regenerate:
+        print('Overwriting existing observations because of --regenerate flag')
```

Comment:
Oh, good call - we may want to regenerate if something went wrong the first time, and this minimizes having to manually delete.